src/mlpack/core/data/string_algorithms.hpp

```diff
@@ -0,0 +1,104 @@
+/** 
+ * @file core/data/string_algorithms.hpp
+ * @author Gopi M. Tatiraju
+ *
+ * Utility fucntions related to string manipulation
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_STRING_ALGORITHMS_HPP
+#define MLPACK_CORE_DATA_STRING_ALGORITHMS_HPP
+
+namespace mlpack{
+namespace data{
+
+/**
+ * A simple trim fucntion to strip off whitespaces 
+ * from both the side of string. If input is a string
+ * with all spaces then str will be empty string.
+ *
+ * @param str string to be trimmed
+ */
+inline void trim(std::string& str)
+{
+  if(str.find_first_not_of(' ') == std::string::npos)
```

Comment:
I think that we need to check for any character in `std::isspace()`, not just `' '`.  Whitespace also includes tab characters.  So, you might want to skip this check and see if you can adapt the loops below to work fine even when all characters are whitespace. :+1: